RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,275 @@
+extending the engine pipeline with a JSON to object converter.
+
+Custom output filters can be written to take regular or irregular output and convert it to json or objects.
+The output filter would also need to have the parameters in use for the original native application.
+This would enable a single filter to be used with any combination of parameters.
+If these custom filters could be added automatically to the invocation of the native app,
+the experience for the user would again be greatly improved.
+
+for example:
+
+```powershell
+PS> MyCustomApp.exe -p1 MyString -p2 42 -p3 resource=blue
+```
+
+_and_ if `MyCustomApp-json` is available as a conversion tool for `MyCustomApp.exe`,
+we can automatically ensure that all the parameters which were used for `MyCustomApp.exe` are passed to the converter.
+In this way, the converter has the exact way `MyCustomApp.exe` was invoked,
+and can be composed to have different behaviors based on the parameters used by `MyCustomApp.exe`.
+
+> **_Implementation note:_** My current prototype searches for an "<app>-json" when a native `ApplicationInfo`
+> is being created during Command Discovery and if found, captures this information in the `ApplicationInfo` object.
+> When the Pipeline processor is being constructed, the extension is checked and if found the "<app>-json" is inserted
+> into the pipeline immediately following the original native application.
+
+The proposed architecture makes it fairly trivial to add an adapter which only calls `ConvertFrom-Json`.
+The combination of this feature, the follow-on work to automatically adding parameters to native applications,
+and our formatting system could change this:
+
+```powershell
+PS> Docker image ls --format='{{json .}}' | ConvertFrom-Json | Where-Object {$_.Repository -match "microsoft"}| Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+4bcdf53ee67b mcr.microsoft.com/powershell           latest                  339MB  2022-09-01 11:34:40 -0700 PDT
+```
+
+to this:
+
+```powershell
+PS> docker image ls | Where-Object Repository -match "microsoft" | Format-Table Id,Repository,Tag,Size,CreatedAt
+
+ID           Repository                             Tag                     Size   CreatedAt
+--           ----------                             ---                     ----   ---------
+4e9617ada198 mcr.microsoft.com/powershell           preview-7.3-mariner-2.0 235MB  2022-09-01 11:48:17 -0700 PDT
+4bcdf53ee67b mcr.microsoft.com/powershell           latest                  339MB  2022-09-01 11:34:40 -0700 PDT
+```
+
+#### History
+
+It's not clear whether history should be altered to show the added pipeline elements.
```

Comment:
This is no longer a problem though with the new feedback provider proposal right?